hat/app/org/hatdex/hat/she/service/FunctionService.scala

```diff
@@ -134,18 +134,24 @@ class FunctionService @Inject() (
       .map(_.get)
   }
 
-  def run(configuration: FunctionConfiguration, startTime: Option[DateTime])(implicit hatServer: HatServer): Future[Done] = {
+  def run(configuration: FunctionConfiguration, startTime: Option[DateTime], useAll: Boolean)(implicit hatServer: HatServer): Future[Done] = {
     val executionTime = DateTime.now()
     logger.info(s"[${hatServer.domain}] SHE function [${configuration.id}] run @$executionTime (previously $startTime)")
     functionRegistry.get[FunctionExecutable](configuration.id)
       .map { function: FunctionExecutable =>
         val fromDate = startTime.orElse(Some(DateTime.now().minusMonths(6)))
         val untilDate = Some(DateTime.now().plusMonths(3))
+        val dataBundleTimeFilter = if (useAll) {
```

Comment:
I don't know if it is an improvement or not more of an FYI (i only just recently found this and in some cases it definitely reads better not certain in this case)

```Option.when(useAll)(configuration.status.lastExecution).flatten ```

Reply:
🤯 

Can you elaborate on what that does in essence?

Reply:
create an option and set it to some when the condition (passed in .when) is true otherwise it is none
in this case it will create a Some(configuration.status.lastExecution) if useAll is true
since configuration.status.lastExecution is already a some it will create Some(Option) so the .flatten will reduce this

Reply:
Looks like a smart solution, my worry would be readability - it's a few extra transformations and it uses `.when` which I find to be a bit less common function.

Reply:
I agree i wish the libraries would standardize on this (Either is .cond) I don't think either is perfect and both are viable solutions. Particularly for variable assignment i like keeping things on a single line but putting if else on a single line (ternary) is not perfect either. This was more an FYI